src/components/ConversationSettings/ConversationAvatarEditor.vue

```diff
@@ -268,24 +268,17 @@ export default {
 		},
 
 		async handleFileChoose(nodes) {
-			const path = nodes[0]?.path
-			if (!path) {
+			const fileid = nodes[0]?.fileid
+			if (!fileid) {
 				return
 			}
 
 			this.loading = true
 			try {
-				const { data } = await axios.post(generateUrl('/avatar'), { path })
-				if (data.status === 'success') {
-					this.loading = false
-				} else if (data.data === 'notsquare') {
-					const tempAvatar = generateUrl('/avatar/tmp') + '?requesttoken=' + encodeURIComponent(getRequestToken()) + '#' + Math.floor(Math.random() * 1000)
-					this.$refs.cropper.replace(tempAvatar)
-					this.showCropper = true
-				} else {
-					showError(data.data.message)
-					this.cancel()
-				}
+				const tempAvatar = generateUrl(`/core/preview?fileId=${fileid}&x=512&y=512&a=1`)
+				this.loading = false
+				this.$refs.cropper.replace(tempAvatar)
+				this.showCropper = true
```

Comment:
Do we still need loading flags here?

Reply:
Looks like a no, we don't wait for server response here (even for preview load)